### PR TITLE
Add check_secret for checking service restarts

### PR DESCRIPTION
**Pull request by zados**
check_secret checks if a service was restarted after secret rotation. The secret rotation is found in a timestamp file.

### CODE REVIEW

File: src/check_secret.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    Gets restart time of service via systemctl
+
+    Args:
+        service_name: Systemd Service name
+
+    Returns: Datetime.
+    """
+    date_s = check_output(
+        f"systemctl show {service_name}.service -p ExecMainStartTimestamp",
+        shell=True).decode("utf-8").split('=')[1].rstrip('\n')
```
Comment (TuxPowered42) [inline src/check_secret.py:53]: Please don't run things with `shell=True`.

File: src/check_secret.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    """
+    Gets restart time of service via systemctl
+
+    Args:
+        service_name: Systemd Service name
+
+    Returns: Datetime.
+    """
+    date_s = check_output(
+        f"systemctl show {service_name}.service -p ExecMainStartTimestamp",
```
Comment (TuxPowered42) [inline src/check_secret.py:52]: Please don't run things with shell=True.

File: src/check_secret.py (hunk 1 above)
Comment (TuxPowered42) [inline src/check_secret.py:53]: To get the value only use `--value`, then you don't need to strip.

File: src/check_secret.py (hunk 2 above)
Comment (TuxPowered42) [inline src/check_secret.py:52]: Please don't run things with shell=True.

File: src/check_secret.py (hunk 1 above)
Comment (TuxPowered42) [inline src/check_secret.py:53]: To get the value only use --value, then you don't need to strip.

**Review by TuxPowered42 — CHANGES_REQUESTED** [review]

File: src/check_secret.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+# THE SOFTWARE.
+
+from argparse import ArgumentParser, Namespace
+from datetime import datetime, timedelta
+from subprocess import check_output
+from sys import exit
+
+
+def parse_args() -> Namespace:
+    parser = ArgumentParser()
```
Comment (kofrezo) [inline src/check_secret.py:33]: A description would be nice.

File: src/check_secret.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    """
+    Gets restart time of service via systemctl.
+
+    Args:
+        service_name: Systemd Service name
+
+    Returns: Datetime.
+    """
+    date_s = check_output(
+        ['/bin/systemctl', 'show', f"{service_name}.service", '-p',
```
Comment (kofrezo) [inline src/check_secret.py:52]: No need for double quotes here. f-strings work with single quotes as well. Use either consistent.

**Review by kofrezo — APPROVED** [review]
Early approve. Please decide for either double or single quotes and stick to one. I consider the description a nice to have.

**Review by TuxPowered42 — APPROVED** [review]

**Pull request merged**